### ISSUE TITLE
F6 Wishlist

### ISSUE THREAD

**Issue by rvinay88** [zurb/foundation#4710]
- [x] Incorporate Sass 3.3 features such as maps and provide a settings API for all Sass components
 - [x] .editorconfig #4720
 - [ ] Unit tests for Sass https://github.com/ericam/true
 - [ ] Support both em's and rem's using a generic calc function #4633 
 - [ ] Experimental advanced grid system - Singularity, Susy, Zen Grids rely more on mixins, make less assumptions and do not constraint design #4134 
 - [ ] Animation library for use with all js plugins - http://h5bp.github.io/Effeckt.css/
 - [ ] Megamenu - http://geedmo.github.io/yamm3/#
 - [ ] Labels with close buttons (like alerts) - Useful in search contexts
 - [x] Accessibility 

Will add more 
Edit: Added accessibility

**Comment by rnaud** [comment]
Less overwriting of base html elements. I'd rather have a table.table {} definition than a table {} one that makes it hard to use table elements with the default behavior.

Also +1 on no reflow. 

My last one would be making more elements able to load content remotely. Modals can do it but tooltips cannot.

**Comment by gakimball** [comment]
We've gotten a number of requests for date and time pickers. #6176

**Comment by cautionbug** [comment]
@gakimball YESYESYESYES! There is a calendar picker out there that *mostly* works (i've had to do some monkeywrenching with how it handles empty values), but it might be a decent starting point:

https://github.com/jockmac22/foundation_calendar_date_time_picker

**Comment by rafibomb** [comment]
@cautionbug Thanks for the tip! We'll check it out!

**Comment by cautionbug** [comment]
@rafibomb i may set up a fork of it, since it doesn't seem there have been any significant changes in quite a while. i'll commit what i did there & follow up here.

So far, the list is looking really good. Ambitious, well-considered... looking forward to what actually makes it into the road map.

**Comment by anish274** [comment]
is foundation planning for material design? I love zurb but my website needs material design. Any Solution?

**Comment by ryanmargheriti** [comment]
Your website doesnt *need* material design @anish274. Nor do developers *need* another material design theme. 

Use one of these instead: 
- https://fezvrasta.github.io/bootstrap-material-design/
- https://github.com/angular/material

**Comment by anish274** [comment]
@ryanmargheriti but my site is already implemented in zurb and the 1st link is the material version of bootstrap. So if my site is bootstrap I can convert it to material design using above link ( I think not sure).
I am totally unaware of angular.

**Comment by gakimball** [comment]
@anish274 We'll leave creating Material skins for Foundation up to the community.

Material Design is a very structured and opinionated system for building interfaces, while Foundation is just meant to be a baseline for a project. It's not the role of our framework to lay that out for you.

**Comment by dpeters911** [comment]
Agreed. Material design is how one company is branding their apps and ux.

I get a sense that the question was related mobile apps in Cordova, in which case Ionic or Foundation for Apps would be more appropriate for applying Material Design conventions to.

Foundation is just that -- a foundation upon which beautiful architecture is expressed.

---- Geoff Kimball wrote ----

@anish274 We'll leave creating Material skins for Foundation up to the community.

Material Design is a very structured and opinionated system for building interfaces, while Foundation is just meant to be a baseline for a project. It's not the role of our framework to lay that out for you.

---

**Comment by cmoralesweb** [comment]
@anish274 If your site "is foundation" and you have to "convert it" to material design (or bootstrap, or whatever) you have a problem.

**Comment by thierryrene** [comment]
@anish274 try this http://ui.lumapps.com/

**Comment by gakimball** [comment]
We've had a request for Material-style circular buttons: #6217

**Comment by listepo** [comment]
customizable select box with support for searching, tagging, remote data sets, infinite scrolling, and many other highly used options?

**Comment by Phil-A-Delphia** [comment]
How about including a forum like the one used on the Foundation website?

**Comment by darkcyber** [comment]
+1 for admin menu ( admin tools ), like admin LTE and the other.
maybe full width row

**Comment by themestechnology** [comment]
How about integrating Font Awesome icons, it seems that quite a lot of people are running both in there app, it would make CSS a bit tidier and help with things like chevrons for menus and breacurms

**Comment by jr-duboc** [comment]
+1 for admin menu, modular structure and Afix

**Comment by themestechnology** [comment]
How about including a schema / schema.org  and Google's Rich Snippets for better SEO?

ie http://schema.org/SiteNavigationElement
http://schema.org/WebPageElement

I could do a pull request if people think this would be worth while

**Comment by gakimball** [comment]
@themestechnology How can we incorporate these rich data hooks into the framework without any data? Wouldn't that be the job of the developer using the framework?

**Comment by themestechnology** [comment]
Well the demo/examples can include this such as Foundation Breadcrumb can have the correct schema and the < nav class="top-bar" data-topbar role="navigation" > can be marked as a navigation element, etc follow by example

**Comment by jr-duboc** [comment]
@themestechnology I tend to agree with @gakimball : I'd say rich data hooks are the responsibility of the developers using Foundation, even though minimal rich data should be included where it makes sense (such as in the navigation example you mentioned)

**Comment by darkcyber** [comment]
@themestechnology 
+1 for font awesome because lot of icons.

**Comment by themestechnology** [comment]
Well yes @jr-duboc , @gakimball  does have a point that not everything can be included such as content definition but it could be considered that the more commonly used element definitions would help enhance peoples apps with out the need to re-invent a new tec

For example things that might be strongly considered...

Navigation elements: < nav class="top-bar" data-topbar role="navigation" itemscope itemtype="http://schema.org/SiteNavigationElement" >

Sidebars (in example documents) : < aside  role="complementary" itemscope itemtype="http://schema.org/WPSideBar" >

Breadcrumbs: < ul xmlns:v="http://rdf.data-vocabulary.org/" class="breadcrumbs" >
< li typeof="v:Breadcrumb" >
< a href="#" rel="v:url" title="" >< span property="v:title" >#< /span >< /a >
< /li >
< /ul >

Website Page < body class="antialiased hide-extras" itemscope itemtype="http://schema.org/WebSite">

*on some occasions content blocks

**Comment by jr-duboc** [comment]
@themestechnology hmmm good point, perhaps adding a few missing rich data snippets where it makes sense...
 @gakimball

**Comment by adamj88** [comment]
How about using SassDoc (http://sassdoc.com) to document the code base? One thing I find lacking when using the Scss is the lack of inline comments/docs for all functions and mixins,

**Comment by gakimball** [comment]
@adamj88 Yes! We're building a tool that will generate documentation from Hologram, SassDoc, and JSDoc. This will also make it easier to design more detailed documentation for the variables and mixins. (See the [Neat documentation](http://thoughtbot.github.io/neat-docs/latest/) for an example.)

**Comment by erredeco** [comment]
notification system like oh-snap or similar ones?

**Comment by gakimball** [comment]
Joyride's functionality could be separated into two plugins: one for the popover, and one for the feature tour. #6251

**Comment by gakimball** [comment]
Load images in Clearing using data URIs instead of the source attribute. #6244

**Comment by airtonix** [comment]
- [ ] move back to libsass

**Comment by gakimball** [comment]
@airtonix The codebase already supports libsass, although we might remove the Compass template from our CLI, and only distribute templates that use libsass.

**Comment by airtonix** [comment]
@gakimball heh. just a brief weekend fling with ruby.

**Comment by johnhamelink** [comment]
:+1: for no reflow - It's a massive pain to have to run that method in Ember

**Comment by rvinay88** [comment]
Thinking out loud here. I want to be able to extend the concept of modularity to Sass components as well. If I want to just use Foundation's grid alone on my next project, it should be easy to get this component alone. 

Right now, if I try to do this, I need some to import globals, use _functions and it's a pain to separate this component alone. Would be awesome if all functions, mixins and variables required by a component is defined within the same sass file.

**Comment by gakimball** [comment]
If you mean *defining* functions within the file versus just importing them, that's not likely to change. We can't anticipate if a function will ever be used on multiple components or not, so we keep them in a separate file. In Foundation 6 our Sass utilities will be a lot more organized, and also separated into multiple files (e.g., units, maps, colors). It's also better for version control if we keep these things in separate files, which is another reason we're moving away from one big functions file.

The components themselves establish dependencies through imports. So right now `_grid.scss` imports `_global.scss`, which imports `_functions.scss`. If any component in Foundation 5 isn't properly importing its dependencies, then that's a bug and we can fix it.

**Comment by sunhump** [comment]
Improved off canvad
Better looking icons
Simple parallax component

**Comment by anish274** [comment]
some example templates for beginners like Landing Page or Blog or Email layouts.
The structured HTML issues which I faced when I started.

**Comment by gakimball** [comment]
@sunhump What would you like to see improved with off-canvas? Our plan right now is to make the component way more generic, as in just an empty box that slides in and out. It won't have built-in menu styles; instead you'll be able to use the menu bar component, or write your own.

**Comment by sunhump** [comment]
@gakimball Sounds good. On top of my head: ability to make it fixed and also add option to overlay content not just push.

**Comment by erredeco** [comment]
Suggestion: I would like a component that could be called "expander" or "modal expander"
Some examples:

http://tympanus.net/Development/CheckoutConcepts/boxes.html (click the cart)
http://tympanus.net/Development/MorphingSearch/ (click on search)

**Comment by erredeco** [comment]
about media queries: I also vote for an x-small media query around 480px

**Comment by gakimball** [comment]
@erredeco We're looking into lowering all of the breakpoints across the board by around 100 pixels. Right now the upper end of small is awkward (it's often a super wide one-column layout), and the lower end of medium is awkward (it's often a super narrow two-column layout).

We'll have to do some experimentation on existing layouts to figure out what the right numbers are. Luckily, it's way easier to mess with breakpoints in Foundation 6. We started some of this in [Foundation for Apps](http://foundation.zurb.com/apps/docs/#!/media-queries), but it'll have more improvements in Foundation for Sites 6.

**Comment by gakimball** [comment]
Right now Abide overloads the `pattern` attribute to add shortcuts for specific patterns, like colors, CCVs, etc. However, this is problematic if the user has JavaScript enabled, in which case the browser-specific validation kicks in, and `pattern="email"` would just become a regexp for the word "email".

One solution, as suggested in #6501 is to use `data-pattern` instead of `pattern`, to keep the two separate.

**Comment by florianbouvot** [comment]
Use BEM naming convention ?

**Comment by SalahAdDin** [comment]
:+1:

**Comment by aibrean** [comment]
HoverIntent for the top bar or hover delay (like with dropdown buttons).

**Comment by SalahAdDin** [comment]
I want help whit this useful features:
centered-block
help-text
badges
signboard(seems like boostrap panels)
list-groups(maybe)

Are you agreed wit this features or you will have problem wit this features?

Thank you.

**Comment by gakimball** [comment]
@SalahAdDin Help text for forms we're thinking about, but it's such a simple component that we aren't sure if it's worth the extra class. Badges we're doing, and panels we currently have, if that's what you mean by signboard. If you're thinking more like Bootstrap's jumbotron, we aren't doing that.

Could you explain what you mean by centered block and list groups?

**Comment by SalahAdDin** [comment]
I don't understand, why help-text is a simple component?
Oh, Jumbotron is a useless class, jajajajaja, don't worry. Yes, i want means bootstrap panels.
Well, for center block i mean
```css
.center-block {
    display: block;
    margin-left: auto;
    margin-right: auto;
}
```

Well, and where can i see the badges and panels? I am using this class in a project but i don't want use boostrap anymore, but, i need this class very much.

Thanks.

**Comment by SalahAdDin** [comment]
And +1 for https://github.com/Ben348/Foundation-Sidebar :+1:, and I need very urgently this feature.

**Comment by themestechnology** [comment]
I think improvement in the Grid Push/Pull should be looked at, it seems to work well enough for two blocks horizontally with a medium-6-push/medium-6-pull but there does not seem to be anything for swapping two blocks vertically with each other in say mobile view small-12-push/small-12-pull. 

I found this frustrating when I was looking at putting the "top-bar" menu at the top above the site header for mobile size screens.

**Comment by SalahAdDin** [comment]
What's about page-header?
```scss
.page-header{
  padding-bottom:rem-calc(5);
  @include rem(margin, 40px 0 20px);
  border-bottom:1px solid $smoke;
}
```

I think that is very handsome.

**Comment by NiklasBr** [comment]
:+1: for the [Sidebar](https://github.com/Ben348/Foundation-Sidebar), and Flexbox mentioned by some people already.

**Comment by erredeco** [comment]
http://tympanus.net/Development/SidebarTransitions/

**Comment by SalahAdDin** [comment]
:+1: http://tympanus.net/Development/SidebarTransitions/

**Comment by erredeco** [comment]
I would like to add another suggestion, maybe it has been already discussed...
I would like to make the the "dropdown button" component more flexible:
I mean:
-  should be possible to use any Icon or text to open the dropdown
- it should be possible to add icon rotation 

this means that the "dropdown button" should be removed as a "special" component... the only component should be "dropdown"

**Comment by gakimball** [comment]
Abide skips validation on hidden fields, which is problematic if you have fields inside a plugin that hides content, like tabs and accordions. We could add some sort of override mechanic to force fields to validate. #6625

**Comment by ventismith** [comment]
Better structure for radio and checkbox lists than just row>column+column as it's overly clunky.

You can hack block-grid to work, but something intentional would be better.

**Comment by collimarco** [comment]
+1 For an autocomplete / typeahead (that can query a remote API). It is an *UI element* that I use in most of my projects and I always have to develop custom solutions or add other heavy libraries only for that (e.g. jQuery UI).

**Comment by mediabeastnz** [comment]
Does anyone find writing rem-calc() is tedious? Can this function be shortened somehow? maybe to just remc() or rc()?

I use this as a px replacement so typing it (atom does a decent job at auto complete) gets a bit old.

**Comment by erredeco** [comment]
http://codepen.io/rgg/pen/waEYye
nothing really new here... but maybe nice for inspiration

**Comment by darkwoods-** [comment]
I would love to see a better top-bar when it comes to customizing it, as the current top-bar can be very painful to customize..

Like for example sometimes I have to do different heights for different media-queries and this messes  up it all.

Or sometimes if I add radius into the top level menu it ends up with effecting the drop downs and messing it all again.

What I'm trying to say here that I don't feel like the current top-bar can be easily customized or it makes you very limited to stay with the default style.

So basically it would be nice to see more SASS customization control over the top-bar by focusing on that there will be in many cases lots of different styles for it on different media queries.

**Comment by mediabeastnz** [comment]
^ what he said!!!

**Comment by SalahAdDin** [comment]
Don't remove Orbit Slider please, is very easy use him, and slick-slider isn't completly, i'm trying make thins that in orbit slider were easey, but, in slick it's very complicated.

**Comment by shivanandwp** [comment]
We should have an option to change css specificity. Perhaps:

    <.mynamespace> {
    …foundation SCSS…
    }

or be able to put in a custom namespace to make foundation CSS selector specific. This will help people like me who just want to download and use foundation inside a WordPress theme.

**Comment by themestechnology** [comment]
@shivanandwp http://foundationpress.olefredrik.com/

**Comment by themestechnology** [comment]
Lots of people asking for things... perhaps the devs at foundation should gather 50 of the best and do a multi vote poll

**Comment by gakimball** [comment]
@shivanandwp This will be possible in Foundation 6 because all CSS exporting is done through mixins.

```scss
#foundation {
  @include foundation-everything; // Exports all CSS
  @include foundation-buttons; // Exports a specific component
}
```

@themestechnology Way ahead of you :) Here's a Trello board with a bunch of feature requests: <https://trello.com/b/QBPKlQqf/foundation-sites>

**Comment by collimarco** [comment]
I would like **a clear, documented way to share Foundation global variables with the Rails application stylesheets**, so that I can reuse the settings in my custom CSS.

At the moment you have to add the following to `application.css`:

```
@import 'foundation_and_overrides';
@import 'a_custom_css_that_uses_foundation_settings';
```

However you cannot use the more proper `= require` of the asset pipeline that keeps the scopes separated.

It would be better to have a file called `foundation_settings` (that contains only the settings and doesn't import the framework) that is `@import`ed by Foundation and by your custom CSS that needs to access the global variables.

**Comment by SalahAdDin** [comment]
Man, what's about the lisft group?
I use this list group, the bootstrap list group for make a elegant list, and use badges, but now, i don't know how adapt this to Foundation.

Are you thinking about it?

**Comment by themestechnology** [comment]
@collimarco 

Copy foundation/scss/foundation/_settings.scss to foundation/scss/

rename it to _custom_settings.scss

Open foundation/scss/foundation.scss at the top of the file add @import "custom_settings";

Now you can set-up your variables using _custom_settings.scss and this will not be effected by updates

**Comment by collimarco** [comment]
@themestechnology That's still a workaround and it's not a viable solution if you use foundation-rails gem.

**Comment by SalahAdDin** [comment]
@gakimball, are you thinking about adapt bootstrap panels to Foundation? I use this in a previous version of my project and have adapt it to Foundation, foundation panels isn't useful as bootstrap panels, and i had to use pricing tables for adapt it.

And list group, i mean, boostrap list groups, Foundation haven't any like this and i think that are a beautiful and elegant solution for a index page.

Help-text will be useful for people that need put a help-text class because his backend don't put place holders in the forms, as me, i work with django and wagtail, and wagtail don't put placeholders in form's inputs. It's very simple do it.

Center-block i used one time, so, i think that will be a helper class but not important class.

I think that will be great if you'll implement a multiple color panels, as boostrap, will be very useful, 
to put different panels depending on the context you want; maybe will be great put multiple colors to pricing tables.

I have been working hard to adapt my project and I compared Foundation, and put here the ideas that came to me. I want to share more ideas and know what you think of them and speedily Foundation 6 as Bootstrap 4 is soon to come.

**Comment by zurbrandon** [comment]
Hey All!
Thanks so much for the amazing feedback on what you are interested in seeing for Foundation 6. The team has been following, responding and acting on this list and we're really excited to be pushing out Foundation 6 pretty shortly. While not every item was done from this we used it as an invaluable roadmap for the goals of the framework. If Foundation 6 launches without a feature you suggested and still want to voice that please feel free to speak up, there's a chance we implemented it differently for a reason or a chance it simply slipped through the cracks. 

Super excited for Foundation 6 Launch and really appreciate the Foundation Community. 

Thankyousomuch,
Brandon (Foundation Team Lead)

**Comment by jarrydfillmore** [comment]
These are awesome suggestions :)

**Comment by dsandstrom** [comment]
Was min/max added to Abide's number validation (https://github.com/zurb/foundation-sites/issues/4059)?  It's supported by HTML5 validation, so it seems like Abide should handle it also.  For now, I will make a custom pattern.
